### PR TITLE
Draft: Setup eval pipeline

### PR DESCRIPTION
**Pull request by bethanyconnolly**
Dont merge, just a draft

### CODE REVIEW

File: experiments/scripts/stability-cluster/run_eval.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# set workdir
+CHEMNLP_PATH=/fsx/proj-chemnlp/$1/chemnlp
+
+# create environment
+source $CHEMNLP_PATH/experiments/scripts/stability-cluster/env_creation_eval.sh $1 $2
+
+# trigger run
+cd $CHEMNLP_PATH/lm-eval2
+python main.py \
+    --model hf-causal \
```
Comment (jackapbutler) [inline experiments/scripts/stability-cluster/run_eval.sh:31]: Could we make these external arguments (or configuration files) that get consumed by the `sbatch` command so we don't need to edit it in between evaluation runs? Similar to the way [this script](https://github.com/jackapbutler/chemnlp/blob/add-hf-mlm-training/experiments/scripts/sbatch_train_hf.sh) doesn't know what configuration / model it is going to be training.

File: experiments/scripts/stability-cluster/run_eval.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    --model_args pretrained=EleutherAI/pythia-160m \
+    --tasks arc_easy \
+    --device 0 \
+    --wandb_log True \
```
Comment (jackapbutler) [inline experiments/scripts/stability-cluster/run_eval.sh:35]: same story for the wandb, personally I think it's more reusable if those lower level arguments get abstracted a level up and can be passed directly when submitting a job.

File: experiments/scripts/stability-cluster/run_eval.sh (hunk 1 above)
Comment (jackapbutler) [reply experiments/scripts/stability-cluster/run_eval.sh:31]: Personally I think it's more reusable if those lower level arguments get abstracted a level up and can be passed directly when submitting a job.

File: experiments/scripts/stability-cluster/env_creation_eval.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# Python requirements
+## cd into your directory inside of proj-chemnlp
+cd /fsx/proj-chemnlp/$2
+
+## clone + submodules (ok if exists)
+[ ! -d 'chemnlp' ] && git clone --recurse-submodules --remote-submodules git@github.com:OpenBioML/chemnlp.git
+
+## install
+conda install pytorch torchvision torchaudio pytorch-cuda=${CUDA_VERSION} -c pytorch -c nvidia --verbose
+pip install -e chemnlp/lm-eval2
```
Comment (maw501) [inline experiments/scripts/stability-cluster/env_creation_eval.sh:29]: Perhaps don't install in editable mode.

File: experiments/scripts/stability-cluster/env_creation_eval.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+conda create --force --prefix ${CONDA_ENV_PATH} python=${PYTHON_VER} -y
+conda activate ${CONDA_ENV_PATH}
+
+# Python requirements
+## cd into your directory inside of proj-chemnlp
+cd /fsx/proj-chemnlp/$2
+
+## clone + submodules (ok if exists)
+[ ! -d 'chemnlp' ] && git clone --recurse-submodules --remote-submodules git@github.com:OpenBioML/chemnlp.git
+
```
Comment (maw501) [inline experiments/scripts/stability-cluster/env_creation_eval.sh:26]: We don't need `--remote-submodules`. In fact, it should be removed from all scripts.

**Review by maw501 — CHANGES_REQUESTED** [review]

File: experiments/scripts/stability-cluster/env_creation_eval.sh (hunk 3 above)
Comment (bethanyconnolly) [reply experiments/scripts/stability-cluster/env_creation_eval.sh:29]: Maybe I will keep in editable mode while we are still in development stage and then we can change it back when we have finished making all changes?

File: experiments/scripts/stability-cluster/env_creation_eval.sh (hunk 4 above)
Comment (bethanyconnolly) [reply experiments/scripts/stability-cluster/env_creation_eval.sh:26]: It is removed now

File: experiments/scripts/stability-cluster/run_eval.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+# set workdir
+CHEMNLP_PATH=/fsx/proj-chemnlp/$1/chemnlp
+
+# create environment
+source $CHEMNLP_PATH/experiments/scripts/stability-cluster/env_creation_eval.sh $1 $2
+
+# trigger run
+cd $CHEMNLP_PATH/lm-eval2
+python main.py $3
```
Comment (maw501) [inline experiments/scripts/stability-cluster/run_eval.sh:30]: Are we using the `main.py` script for this repo or creating a different version per comments [here](https://github.com/OpenBioML/lm-eval2/pull/1) (I know this is for the `lm-eval2` repo)? If so this file reference will need updating.

**Pull request merged**